tests/MongoDB.Driver.Tests/EncryptionTests.cs

```diff
@@ -0,0 +1,174 @@
+                    var cryptClient2 = libMongoCryptController2._cryptClient();
+                    var areTheSame = object.ReferenceEquals(cryptClient1, cryptClient2);
+                    areTheSame.Should().BeTrue();
+                }
+                else
+                {
+                    libMongoCryptController1.Should().BeNull();
+                    libMongoCryptController2.Should().BeNull();
+                }
+
+                var kmsProviders = GetKmsProviders();
+                var clientEncryption = GetClientEncryption((MongoClient)client1.Wrapped, kmsProviders);
+                clientEncryption._libMongoCryptController()._cryptClient().Should().NotBeNull();
+            }
+        }
+
+        [SkippableTheory]
+        [ParameterAttributeData]
+        public void Mongocryptd_should_be_initialized_for_only_for_auto_encryption([Values(false, true)] bool withAutoEncryption)
+        {
+            RequireServer.Check().Supports(Feature.ClientSideEncryption);
+
+            using (var disposableClient = GetClient(withAutoEncryption))
+            {
+                var kmsProviders = GetKmsProviders();
+                var client = (MongoClient)disposableClient.Wrapped;
+                var clientEncryption = GetClientEncryption(client, kmsProviders);
+                if (withAutoEncryption)
+                {
+                    client.LibMongoCryptController.Should().NotBeNull();
+                    var clientController = client.LibMongoCryptController;
+                    clientController._mongocryptdClient().Should().NotBeNull();
+
+                    var clientEncryptionController = clientEncryption._libMongoCryptController();
+                    clientEncryptionController.Should().NotBeNull();
+                    var mongoCryptD = clientEncryptionController._mongocryptdClient();
+                    mongoCryptD.Should().BeNull();
+                }
+                else
+                {
+                    client.LibMongoCryptController.Should().BeNull();
+
+                    var clientEncryptionController = clientEncryption._libMongoCryptController();
+                    clientEncryptionController.Should().NotBeNull();
+                    var mongoCryptD = clientEncryptionController._mongocryptdClient();
+                    mongoCryptD.Should().BeNull();
+                }
+            }
+        }
+
+        private DisposableMongoClient GetClient(bool withAutoEncryption = false, Dictionary<string, object> extraOptions = null)
+        {
+            var mongoClientSettings = new MongoClientSettings();
+
+            if (withAutoEncryption)
+            {
+                if (extraOptions == null)
+                {
+                    extraOptions = new Dictionary<string, object>()
+                    {
+                        { "mongocryptdSpawnPath", Environment.GetEnvironmentVariable("MONGODB_BINARIES") ?? string.Empty }
```

Comment:
Nit: I think the standard path should also work, based off what I've seen in the scripts, but since this clearly also works, I'm fine with it.